### ISSUE TITLE
Unable to override inputs in dependent profiles

### ISSUE THREAD

**Issue by g1ps** [inspec/inspec#4354]
### 🎛 Description

**Choose one:** is this a 🐛 bug report or 🙋 feature request? Don't know.

We are unable to override inputs in a dependent profile. Both parent and child profiles use 
```
params = yaml(content: inspec.profile.file('validation.yml')).params
var = params['var']
...
```
to read inputs from a file in the local `files` directory. None of the methods described in the docs work i.e. declaring inputs in inspec.yml metadata (yes, we specified the child profile's name as req'd for V4), passing the file via `--input-file`, etc. The child profile always reads its inputs from the child `validation.yml` file. Is it possible to override inputs set this way?

We have dozens of variables set in `validation.yml` so it doesn't seem practical to use the multi-line metadata approach. What is the recommended approach for this use case?

Thanks.

### 🌍 InSpec and Platform Version

```
E:\temp\inspec>inspec --version
4.7.24
```

### 🤔 Replication Case

Create a parent/child profile relationship wihere both profiles import inputs as decribed above.

### 💻 Stacktrace

No errors but nothing is overridden.

**Comment by clintoncwolfe** [comment]
Hi @g1ps, I'd like to help, but I'm unclear what you're reporting - you're asking for a behavior change on *inputs*, but your example posted is about the `yaml` resource and the profile file reader facility, neither of which are related to inputs. Note also that the profile files facility does not use inheritance (#3030) - so yes, a profile file in a child profile will always have that content. It looks like you've posted a workaround or partial workaround.

Do you think you could add a simplified example showing exactly what it is you are doing, what you expect to see, and what you see instead?

**Comment by g1ps** [comment]
I'm lost as to what you could fail to understand. At no point did I ask for any change. I haven't posted anything vaguely resembling a workaround. Where did you get that from? The docs clearly show inputs used in profiles and discuss the inheritance rules. We haven't used `input` statements, I don't think they were around when the profile was written, we used the yaml import statements shown above. Given that both the parent and child profiles use this method, we are unable to override variables declared in the child, when calling it from the parent.

The docs also clearly describe overriding values by passing a yaml file on the command lie:
```
Setting Input values using --input-file

You may also provide inputs and values via YAML files on the command line. ...
```
I showed that we tried that and it fails as well. The docs discuss, at some length, how profile inheritance is supposed to work but we can't make it so. 
```
In addition, for inherited profiles:

dependent profile metadata < wrapper profile metadata
```
so how do you arrive at "the profile files facility does not use inheritance"??

So, to try and make it even clearer, I guess my question is two-fold:
1) How do we get inheritance to work so that we can override values set in the child profile?
2) Is the current failure a results of using the yml importation method we chose when we wrote this?

If the answer to 2) is yes I'm happy to refactor it, I'm just trying to understand what's required to get this to work. We need to separate config. from code and to be able to change the input values, i.e. the values passed to the profile, on demand.

 Thanks.

**Comment by clintoncwolfe** [comment]
Please keep in mind I'm on your side; I want to see you successful in your use of InSpec.

I think you may have some confusion among three separate concepts, which are unrelated.

* [InSpec Inputs](https://www.inspec.io/docs/reference/inputs) are a feature which allow you to read in data from a variety of sources, including profile metadata, special plugins, or -confusingly in this case - datafiles specified on the command line which happen to be YAML-formatted. Inputs have a variety of features, including some form of inheritance and profile namespacing. The only way to access these values in profile code is to use the `input()` DSL call in your profiles. So, if you are not calling `input()`, you are not using Inputs. Conversion to using Inputs is probably what you want. 

* [the YAML resource](https://www.inspec.io/docs/reference/resources/yaml) is an InSpec resource that simply reads and parses a YAML file you specify, and returns the data for you to use however you like. As InSpec has no idea where the data came from (you specify the path), the resource doesn't exhibit any inheritance behavior. This is just a YAML parser.

* [profile files](https://www.inspec.io/docs/reference/profiles/#profile-files) are a feature which allow you deliver files of your choosing along with a profile. InSpec doesn't know what those files are, semantically, and so doesn't provide inheritance for them (arguably it should - see #3030). 

When I talk about Inputs, I'm exclusively talking about the first meaning. InSpec profiles are really just Ruby - you could get data into them in any number of ways, including (as you have done here) using the YAML resource in conjunction with the file profile files feature. But that should not lead to any expectation of being able to override values - you've built a (very clever) way of doing something that looks a lot like Inputs, but actually isn't.

So, if you want it to work like Inputs, and have documented behavior like Inputs, you should use `input()` and use the input system. It should work - or hey, you may find bugs! Happy to help narrow that down and hopefully get them fixed.

**Comment by g1ps** [comment]
Hi,

Sorry, I didn't believe otherwise. :-)

OK, part of what you descibed I have since discovered. I've refactored so that the base profile uses inputs in the metadata and overrides now work as expected.  I suspected that our way of importing the config. file was the problem and indeed it was. I was only ever talking about the first meaning, too, so I'm still not quite sure what you're driving at with part 3 there but they do inherit inputs, as per the docs, and I have that working.

Thanks.
